README.md

```diff
@@ -280,6 +280,110 @@ run --arg value1 --arg value2
 # => values = [value1, value2]
 ```
 
+#### `@Command.Rest({required?: number})`
+
+Specifies that the command accepts a set of positional arguments. By default, no arguments are required, but this can be changed by setting the `required` option.
+
+```ts
+class RunCommand extends Command {
+    @Command.Rest()
+    public values: string[];
+    // ...
+}
+```
+
+Generates:
+
+```bash
+run value1 value2
+# => values = [value1, value2]
+
+run value1
+# => values = [value1]
+
+run
+# => values = []
+```
+
+**Note:** Rest arguments are strictly positional. All options found between rest arguments will be consumed as options of the `Command` instance. For proxying, `Command.Proxy` has to be used instead.
+
+**Note:** Rest arguments can be surrounded by other *finite* *non-optional* positionals such as `Command.String({required: true})`.
+
+**Advanced Example:**
+
+```ts
+class CopyCommand extends Command {
+    @Command.Rest({required: 1})
+    sources: string[] = [];
+
+    @Command.String()
+    destination!: string;
+
+    @Command.Boolean(`-f,--force`)
+    force: boolean = false;
+
+    @Command.String(`--reflink`, {tolerateBoolean: true})
+    reflink: string | boolean = false;
+
+    // ...
+}
+```
+
+Generates:
+
+```bash
+run src dest
+# => sources = [src]; destination = dest
+
+run src1 src2 dest
+# => sources = [src1, src2]; destination = dest
+
+run src1 --force src2 dest
+# => sources = [src1, src2]; destination = dest; force = true
+
+run src1 src2 --reflink=always dest
+# => sources = [src1, src2]; destination = dest; reflink = always
+
+run src
+# => Error - Not enough positional arguments.
+
+run dest
+# => Error - Not enough positional arguments.
+```
+
+#### `@Command.Proxy({required?: number})`
+
+Specifies that the command accepts an infinite set of positional arguments that will not be consumed by the options of the `Command` instance. Used to proxy arguments to another command. By default, no arguments are required, but this can be changed by setting the `required` option.
+
+```ts
+class RunCommand extends Command {
+    @Command.Proxy()
+    public args: string[];
+    // ...
+}
+```
+
+Generates:
+
+```bash
+run value1 value2
+# => values = [value1, value2]
+
+run value1 --foo
+# => values = [value1, --foo]
+
+run --bar=baz
+# => values = [--bar=baz]
+```
+
+**Note:** Proxying can only happen once per command. Once triggered, a command can't get out of "proxy mode", all remaining arguments being proxied into a list. "Proxy mode" can be triggered in the following ways:
+
+- By passing a positional or an option that doesn't have any listeners attached to it. This happens when the listeners don't exist in the first place.
+
+- By passing a positional that doesn't have any *remaining* listeners attached to it. This happens when the listeners have already consumed a positional.
```

Comment:
We probably shouldn't use the concept of listener in the documentation, as it doesn't match the API that most people will consume.

Reply:
I've used "listener" because that's what was used inside the [JSDoc comments](https://github.com/arcanis/clipanion/blob/9a6700b740726ac83e58115d4b210dcf3cea1e94/sources/advanced/Command.ts#L216) attached to the decorators. Do you have any better word ideas?